packages/create-app/src/generate/generateNpmRc.ts

```diff
@@ -21,7 +21,12 @@ export function generateNpmRc({
   osdkPackage: string;
   osdkRegistryUrl: string;
 }): string {
-  const withoutProtocol = osdkRegistryUrl.replace(/^https:\/\//, "");
+  // pnpm requires a trailing slash in .npmrc
+  // https://github.com/pnpm/pnpm/issues/5941
+  const withTrailingSlash = osdkRegistryUrl.endsWith("/")
+    ? osdkRegistryUrl
+    : osdkRegistryUrl + "/";
+  const withoutProtocol = withTrailingSlash.replace(/^https:\/\//, "");
   return `//${withoutProtocol}:_authToken=\${FOUNDRY_TOKEN}\n`
-    + `${osdkPackage.split("/")[0]}:registry=${osdkRegistryUrl}\n`;
+    + `${osdkPackage.split("/")[0]}:registry=${withTrailingSlash}\n`;
```

Comment:
minor nit: I know this existed from before, but we can replace this with a descriptive name to fit with the rest

```
const packageName = osdkPackage.split("/")[0];
```
```suggestion
    + `${packageName}:registry=${withTrailingSlash}\n`;
```